oak-run-elastic/src/main/java/org/apache/jackrabbit/oak/index/ElasticIndexCommand.java

```diff
@@ -0,0 +1,201 @@
+                indexerSupport.getCheckpointInfo());
+
+        if (opts.getCommonOpts().isMongo() && idxOpts.isDocTraversalMode()) {
+            log.info("Using Document order traversal to perform reindexing");
+            try (ElasticDocumentStoreIndexer indexer = new ElasticDocumentStoreIndexer(indexHelper, indexerSupport, indexOpts.getIndexPrefix(),
+                    indexOpts.getElasticScheme(), indexOpts.getElasticHost(),
+                    indexOpts.getElasticPort(), indexOpts.getApiKeyId(), indexOpts.getApiKeySecret())) {
+                indexer.reindex();
+                // Wait for default flush interval before exiting the try block
+                // to make sure the client is not closed before the last flush
+                // TODO : See if this can be handled in a better manner
+                Thread.sleep(ElasticIndexDefinition.BULK_FLUSH_INTERVAL_MS_DEFAULT*2);
+            } catch (InterruptedException e) {
+                //
+            }
+        } else {
+            try(ElasticOutOfBandIndexer indexer = new ElasticOutOfBandIndexer(indexHelper, indexerSupport, indexOpts.getIndexPrefix(),
+                    indexOpts.getElasticScheme(), indexOpts.getElasticHost(),
+                    indexOpts.getElasticPort(), indexOpts.getApiKeyId(), indexOpts.getApiKeySecret())) {
+
+                indexer.reindex();
+                // Wait for default flush interval before exiting the try block
+                // to make sure the client is not closed before the last flush
+                Thread.sleep(ElasticIndexDefinition.BULK_FLUSH_INTERVAL_MS_DEFAULT*2);
+            } catch (InterruptedException e) {
+                //
+            }
+        }
+        indexerSupport.writeMetaInfo(checkpoint);
+    }
+
+    private IndexerSupport createIndexerSupport(IndexHelper indexHelper, String checkpoint) {
+        IndexerSupport indexerSupport = new IndexerSupport(indexHelper, checkpoint);
+
+        File definitions = indexOpts.getIndexDefinitionsFile();
+        if (definitions != null) {
+            Preconditions.checkArgument(definitions.exists(), "Index definitions file [%s] not found", getPath(definitions));
+            indexerSupport.setIndexDefinitions(definitions);
+        }
+        return indexerSupport;
+    }
+
+    static Path getPath(File file) {
+        return file.toPath().normalize().toAbsolutePath();
+    }
+
+    private void shutdownLogging() {
+        LoggingInitializer.shutdownLogging();
+    }
+
+    private static void logCliArgs(String[] args) {
+        log.info("Command line arguments used for indexing [{}]", Joiner.on(' ').join(args));
+        List<String> inputArgs = ManagementFactory.getRuntimeMXBean().getInputArguments();
+        if (!inputArgs.isEmpty()) {
+            log.info("System properties and vm options passed {}", inputArgs);
+        }
+    }
+
+    public static void setDisableExitOnError(boolean disableExitOnError) {
```

Comment:
Actually I used this from the lucene index command - this can be used if we use this command from some external script. For the time being, we can just let it be.